vector/src/main/java/im/vector/app/features/poll/create/CreatePollViewModel.kt

```diff
@@ -81,12 +105,20 @@ class CreatePollViewModel @AssistedInject constructor(
                 _viewEvents.post(CreatePollViewEvents.NotEnoughOptionsError(requiredOptionsCount = MIN_OPTIONS_COUNT))
             }
             else                                     -> {
-                room.sendPoll(state.question, nonEmptyOptions)
+                when (state.mode) {
+                    PollMode.CREATE -> room.sendPoll(state.pollType, state.question, nonEmptyOptions)
+                    PollMode.EDIT   -> sendEditedPoll(state.editedEventId!!, state.pollType, state.question, nonEmptyOptions)
+                }
                 _viewEvents.post(CreatePollViewEvents.Success)
             }
         }
     }
 
+    private fun sendEditedPoll(editedEventId: String, pollType: PollType, question: String, options: List<String>) {
+        val editedEvent = room.getTimeLineEvent(editedEventId) ?: return
+        room.editPoll(editedEvent, pollType, question, options)
```

Comment:
Nit: `room.editPoll()` first parameter could be an `eventId: String` to let the SDK user have a nicer API.

Reply:
But we don't have the `room` instance in `EventEditor`. Do we really want to inject `session` and pass the `roomId` and `eventId`?

Reply:
OK, the API is similar to the other ones, you can leave it like this, sorry.